sdk/identity/azure-identity/src/main/java/com/azure/identity/IntelliJCredential.java

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.core.credential.AccessToken;
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.credential.TokenRequestContext;
+import com.azure.identity.implementation.IdentityClient;
+import com.azure.identity.implementation.IdentityClientBuilder;
+import com.azure.identity.implementation.IdentityClientOptions;
+import com.azure.identity.implementation.MsalToken;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+/**
+ * A credential provider that provides token credentials from Azure Tools for IntelliJ plugin credential cache.
+ *
+ * <p> If the developer has authenticated successfully with Azure Tools for IntelliJ plugin in the IntelliJ IDE then
+ * this credential can be used in the development code to reuse the cached plugin credentials.</p>
+ *
+ * @see IntelliJCredentialBuilder
+ */
+@Immutable
+public class IntelliJCredential implements TokenCredential {
```

Comment:
This class should not be public, we're just adding the IDE style credentials to the `DefaultAzureCredential` but not actually exposing the types directly.

Reply:
internalized.